### PR TITLE
feat(arns): add API to block and unblock ArNS names

### PR DESCRIPTION
**Pull request by karlprieb**

### CODE REVIEW

File: src/middleware/arns.ts
Comment (karlprieb) [inline src/middleware/arns.ts:121]: @djwhitt Do you think this is the best place to have this class?

I have some questions about this class:
 1. If the cache is updated from the db when it starts and names are added and removed when we block and unblock names I don't think we need TTL and the setInterval to update the cache from the db.
 2. If 1 is correct we don't even need this class as the cache can be just a variable.

File: src/routes/ar-io.ts
Comment (karlprieb) [inline src/routes/ar-io.ts:167]: Should we just run `db.unblockName` from the cache itself and just call `removeName` here?

File: src/routes/ar-io.ts
Comment (karlprieb) [reply src/routes/ar-io.ts:152]: Great suggestion!

File: src/routes/ar-io.ts
Comment (karlprieb) [reply src/routes/ar-io.ts:146]: @djwhitt this is a real issue. Maybe we should wait for the db response to update the cache and to respond the api request

Comment (djwhitt) [reply src/routes/ar-io.ts:146]: Agreed. We should wait on the DB operation to complete before updating the cache.

File: test/moderation-schema.sql
Comment (djwhitt) [reply test/moderation-schema.sql:28]: @karlprieb We don't need the foreign key, but we should add the index.

File: src/routes/ar-io.ts
Comment (djwhitt) [reply src/routes/ar-io.ts:167]: I lean towards keeping them separate. That keeps the cache interface simpler.

File: src/routes/ar-io.ts
Comment (djwhitt) [reply src/routes/ar-io.ts:152]: Agreed, let's do this.

File: src/middleware/arns.ts
Comment (karlprieb) [reply src/middleware/arns.ts]: [`95f516a` (#245)](https://github.com/ar-io/ar-io-node/pull/245/commits/95f516a24ea4498d67addb2668035d6f02c17d96)

File: src/routes/ar-io.ts
Comment (karlprieb) [reply src/routes/ar-io.ts]: [`2f02a4a` (#245)](https://github.com/ar-io/ar-io-node/pull/245/commits/2f02a4a48323fccab7e79a6e4804bb599501c594)

File: test/moderation-schema.sql
Comment (karlprieb) [reply test/moderation-schema.sql:28]: [`bf32fd2` (#245)](https://github.com/ar-io/ar-io-node/pull/245/commits/bf32fd2210ae5af5e164d00097110c632af9b43b)

**Review by djwhitt — APPROVED** [review]

**Pull request merged**
